Dockerfile.cuda

```diff
@@ -0,0 +1,79 @@
+# Dockerfile for development purposes.
+# Read docs/development.md for more information
+# vi: ft=dockerfile
+
+################################################################################
+# Weaviate CUDA builder
+# Build stage
+FROM nvcr.io/nvidia/cuda:12.5.1-devel-ubuntu22.04 AS cuvs_server_builder
+
+# For local binary case, directly copy the files from named context
+COPY --from=libs /lib /opt/cuvs/lib/
+COPY --from=libs /include /opt/cuvs/include/
+
+# Install the library using mamba if build context files aren't present
+RUN if [ ! -d "/opt/cuvs/lib" ] || [ -z "$(ls -A /opt/cuvs/lib)" ]; then \
+    echo "Installing from mamba" && \
+    wget -qO- https://micro.mamba.pm/api/micromamba/linux-64/latest | tar -xvj bin/micromamba && \
+    ./bin/micromamba shell init -s bash -p /opt/conda && \
+    mkdir -p /opt/conda/conda-meta && \
+    eval "$(./bin/micromamba shell hook -s bash)" && \
+    mamba install -y -c conda-forge -c nvidia -c rapidsai cuvs && \
+    mkdir -p /opt/lib /opt/include && \
+    cp -r /opt/conda/lib/* /opt/lib/ && \
+    cp -r /opt/conda/include/* /opt/include/ ; \
+    fi
+
+# Install Go 1.22.4
+RUN apt-get update && apt-get install -y wget && \
+    wget https://go.dev/dl/go1.22.4.linux-amd64.tar.gz && \
+    rm -rf /usr/local/go && \
+    tar -C /usr/local -xzf go1.22.4.linux-amd64.tar.gz && \
+    rm go1.22.4.linux-amd64.tar.gz
+
+# Add Go to PATH
+ENV PATH=$PATH:/usr/local/go/bin
+
+COPY . .
+
+# install cgo-related dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    build-essential
+
+ENV CGO_CFLAGS="-I/usr/local/cuda/include -I/opt/cuvs/include"
+ENV CGO_LDFLAGS="-L/usr/local/cuda/lib64 -L/opt/cuvs/lib -lcudart -lcuvs -lcuvs_c"
+
+# RUN sed -i '/#include <time.h>/a #include <asm-generic/errno.h>' /usr/local/go/src/runtime/cgo/gcc_libinit.c
+# ENV CC=x86_64-linux-gnu-gcc
+
+ARG TARGETARCH
+ARG GIT_BRANCH="unknown"
+ARG GIT_REVISION="unknown"
+ARG BUILD_USER="unknown"
+ARG BUILD_DATE="unknown"
+ARG EXTRA_BUILD_ARGS=""
+
+RUN  GOOS=linux GOARCH=$TARGETARCH CGO_ENABLED=1 go build $EXTRA_BUILD_ARGS \
+    -tags cuvs \
+    -ldflags "-w \
+    -X github.com/weaviate/weaviate/usecases/build.Branch=$GIT_BRANCH \
+    -X github.com/weaviate/weaviate/usecases/build.Revision=$GIT_REVISION \
+    -X github.com/weaviate/weaviate/usecases/build.BuildUser=$BUILD_USER \
+    -X github.com/weaviate/weaviate/usecases/build.BuildDate=$BUILD_DATE" \
+    -o /weaviate /cmd/weaviate-server
+
+################################################################################
+# Weaviate CUDA runner (no differentiation between dev/test/prod - 12 factor!)
+FROM nvcr.io/nvidia/cuda:12.5.1-runtime-ubuntu22.04 as weaviate_cuvs
+
+COPY --from=cuvs_server_builder /weaviate /bin/weaviate
+COPY --from=cuvs_server_builder /opt/cuvs/lib /opt/cuvs/lib
+
+ENV LD_LIBRARY_PATH=/opt/cuvs/lib:$LD_LIBRARY_PATH
+
+RUN apt-get update && apt-get install -y ca-certificates openssl
+
+EXPOSE 8080
```

Comment:
We don't have `EXPOSE 8080` in the main Dockerfile so for consistency I'd remove this